ts/Grid/Core/Pagination/Pagination.ts

```diff
@@ -154,149 +241,511 @@ class Pagination {
         this.reflow();
 
         this.contentWrapper = makeHTMLElement('div', {
-            className: Globals.getClassName('paginationWrapper')
+            className: Globals.getClassName('pgWrapper')
         }, this.cell);
 
-        this.renderNavButtons();
+        // Update total pages first to ensure correct calculations
+        this.updateTotalPages();
+
+        // Render all components
+        this.renderPageInfo();
+        this.renderControls();
+        this.renderPageSizeSelector();
+
+        // Update button states after rendering
+        this.updateButtonStates();
     }
 
     /**
-     * Render the navigation buttons.
+     * Render the page information text.
      */
-    public renderNavButtons(): void {
-        const pg = this;
-        const grid = pg.grid;
-        const originalDataTable = grid.dataTable;
+    public renderPageInfo(): void {
+        if (!this.options.controls?.pageInfo) {
+            return;
+        }
 
-        this.prevButton = makeHTMLElement(
-            'button',
-            {
-                innerHTML: '&lsaquo;' // TODO: add lang support
-            },
-            this.contentWrapper
-        );
+        this.pageInfoElement = makeHTMLElement('div', {
+            className: Globals.getClassName('pgPageInfo')
+        }, this.contentWrapper);
 
-        this.setButtonState(
-            this.prevButton,
-            true
-        );
+        this.updatePageInfo();
+    }
 
-        this.prevButton.addEventListener(
-            'click',
-            function (): void {
-                pg.updatePage(false);
-            }
-        );
+    /**
+     * Update the page information text.
+     */
+    public updatePageInfo(): void {
+        if (!this.pageInfoElement) {
+            return;
+        }
 
-        this.nextButton = makeHTMLElement(
-            'button',
-            {
-                innerHTML: '&rsaquo;' // TODO: add lang support
-            },
-            this.contentWrapper
+        const startItem = (this.currentPage - 1) * this.currentPageSize + 1;
+        const endItem = Math.min(
+            this.currentPage * this.currentPageSize,
+            this.totalItems
         );
 
-        this.nextButton.addEventListener(
-            'click',
-            function (): void {
-                pg.updatePage();
-            }
-        );
+        const pageInfoText = this.formatText(this.lang.pageInfo as string, {
```

Comment:
`pageInfo` can be either `string` or `undefined`. Can we check if it's `undefined` before putting it into the `formatText` (or check inside the `formatText`), rather than casting?